web/modules/custom/dpl_article/dpl_article.module

```diff
@@ -0,0 +1,55 @@
+<?php
+
+/**
+ * @file
+ * DPL article module.
+ */
+
+use Drupal\Core\Form\FormStateInterface;
+use Drupal\node\Entity\Node;
+
+/**
+ * Implements hook_preprocess_HOOK().
+ */
+function dpl_article_preprocess_node(array &$variables): void {
+  $node = $variables['node'] ?? NULL;
+
+  // Process only if the node is a valid Node entity.
+  if (!($node instanceof Node)) {
```

Comment:
Please replace not with `NodeInterface`.

In general when doing object oriented programming you often want to code against interfaces instead of implementations.